packages/webpack/src/utils/module-federation/share.ts

```diff
@@ -66,14 +76,59 @@ export function shareWorkspaceLibraries(
         (aliases, library) => ({ ...aliases, [library.name]: library.path }),
         {}
       ),
-    getLibraries: (eager?: boolean): Record<string, SharedLibraryConfig> =>
-      pathMappings.reduce(
-        (libraries, library) => ({
+    getLibraries: (eager?: boolean): Record<string, SharedLibraryConfig> => {
+      let pkgJson = null;
+      if (
+        project?.root &&
+        existsSync(
+          joinPathFragments(workspaceRoot, project.root, 'package.json')
+        )
+      ) {
+        pkgJson = readJsonFile(
+          joinPathFragments(workspaceRoot, project.root, 'package.json')
+        );
+      }
+      return pathMappings.reduce((libraries, library) => {
+        // Check to see if the library version ranged is declared in the root package.json
+        let version =
+          pkgJson?.dependencies?.[library.name] ??
+          pkgJson?.devDependencies?.[library.name] ??
+          null;
+        if (!version) {
+          const workspaceLib = workspaceLibs.find(
+            (lib) => lib.importKey === library.name
+          );
+
+          const workspacePkgJsonPath = join(workspaceLib.root, 'package.json');
+          if (existsSync(workspacePkgJsonPath)) {
+            const workspacePackageJson: PackageJson =
+              readJsonFile(workspacePkgJsonPath);
+
+            if (!workspacePackageJson) {
+              logger.warn(
+                `Could not find a version for "${library.name}" in the root "package.json"`
+              );
+            } else {
+              version = workspacePackageJson.version;
+            }
+          }
+        }
+
+        return {
           ...libraries,
-          [library.name]: { requiredVersion: false, eager },
-        }),
-        {} as Record<string, SharedLibraryConfig>
-      ),
+          [library.name]: {
+            ...(version
+              ? {
+                  requiredVersion: version,
+                  singleton: false,
```

Comment:
I think we still want this to be true, otherwise each remote will use its own copy of the library, which can be breaking for shared libraries that have shared state between remotes. 

Unless the intention is to force the users to declare explicitly in their module-federation.config which libraries should be singletons.

Reply:
yea that's right thanks.